tests/test_logutils.py

```diff
@@ -0,0 +1,141 @@
+     .fit(np.array([1, 1, 1, 2, 2, 2]).reshape(6, 1), [2, 2, 2, 3, 3, 3]))
+
+
+class TestLogUtils(unittest.TestCase):
+
+    def setUp(self):
+        self.TEMP_FILES = []
+        self.TEMP_FILE_PATHS = []
+        self.LOGGERS = []
+
+    def tearDown(self):
+        self.reset()
+
+    def reset(self):
+        for i, temp_file in enumerate(self.TEMP_FILES):
+            temp_file.close()
+            del self.TEMP_FILES[i]
+        for i, temp_file_path in enumerate(self.TEMP_FILE_PATHS):
+            unlink(temp_file_path)
+            del self.TEMP_FILE_PATHS[i]
+        for i, logger in enumerate(self.LOGGERS):
+            close_and_remove_logger_handlers(logger)
+            del self.LOGGERS[i]
+        warnings.showwarning = orig_showwarning
+
+
+    def testGetSKLLLogger(self):
+        self.reset()
+
+        temp_file = NamedTemporaryFile("w", delete=False)
+        temp_file.close()
+        self.TEMP_FILES.append(temp_file)
+        self.TEMP_FILE_PATHS.append(temp_file.name)
+        logger = get_skll_logger("testGetSKLLLogger", temp_file.name)
+        self.LOGGERS.append(logger)
+
+        # Send a regular log message
+        msg1 = "message 1"
+        logger.info(msg1)
+
+        # Send a regular log message
+        msg2 = "message 2"
+        logger.info(msg2)
+
+        with open(temp_file.name) as temp_file:
+            log_lines = temp_file.readlines()
+            assert log_lines[0].endswith("INFO - {}\n".format(msg1))
+            assert log_lines[1].endswith("INFO - {}\n".format(msg2))
+
+
+    def testGetSKLLLoggerWithWarning(self):
+        self.reset()
+
+        temp_file = NamedTemporaryFile("w", delete=False)
+        temp_file.close()
+        self.TEMP_FILES.append(temp_file)
+        self.TEMP_FILE_PATHS.append(temp_file.name)
+        logger = get_skll_logger("testGetSKLLLoggerWithWarning", temp_file.name)
+        self.LOGGERS.append(logger)
+
+        # Send a regular log message
+        msg1 = "message 1"
+        logger.info(msg1)
+
+        # Trigger a couple ``sklearn`` warnings
+        trigger_sklearn_warnings()
+
+        # Send a regular log message
+        msg2 = "message 2"
+        logger.info(msg2)
+
+        with open(temp_file.name) as temp_file:
+            log_lines = temp_file.readlines()
+            assert log_lines[0].endswith("INFO - {}\n".format(msg1))
+            cv_sklearn_warning_re = \
+                re.compile(r"WARNING - [^\n]+site-packages/sklearn/model_selection/_split\.py"
+                           r":2053: FutureWarning:You should specify a value for 'cv' instead"
+                           r" of relying on the default value\. The default value will change"
+                           r" from 3 to 5 in version 0\.22\.")
+            assert cv_sklearn_warning_re.search("".join(log_lines[1:-1]))
+            n_estimators_sklearn_warning_re = \
+                re.compile(r"WARNING - [^\n]+site-packages/sklearn/ensemble/forest\.py:246: "
```

Comment:
I liked everything you did, but can this be some other warning rather than future warning. If this warning is removed or the functionality is absolute later on future, then we have to re-work to fix the test case too, isn't it?

Reply:
Yeah, that's a good point. Can you think of a case that raises a warning from `sklearn`? I have been trying to find a case to use and this was that worked out. But, yes, it will be a problem in the future.

Reply:
Let me search if I can find something.